### ISSUE TITLE
Unecessary break before closing parenthesis

### ISSUE THREAD

**Issue by Julow** [ocaml-ppx/ocamlformat#793]
When the expression grows, formatting change:

```ocaml
let _ = f (fun () -> {|
Multi line
but short
|})

let _ =
  f (fun () ->
      {|
Multi line
with a lot of text
Foooooooooooooooooooooooooooooooooooooooooooooo
Bar
|}
  )
```

**Comment by gpetiot** [comment]
I'm not sure what we can do for this one (except rewriting the source to put the literal strings in a variable).

**Comment by mbarbin** [comment]
I don't understand your comment @gpetiot .  My perspective is that this is a bug.  Do you feel differently?

**Comment by mbarbin** [comment]
Wouldn't it be related to #799 ?

**Comment by gpetiot** [comment]
It doesn't look related to this one to me, but I will have a closer look soon.

**Comment by gpetiot** [comment]
For me it's not a bug, the first example does not break any box, the literal string contains the `\n` character but that doe snot break any box (if we replace `\n` with any other character it fits on a single line), whereas in the second example the literal string is too long to fit on a single line so the box breaks, just like it would for any expression. I don't see what we can do to not have this behavior.

**Comment by Julow** [comment]
We could check for `\n` in the string ? I see this is done for an `is_simple` function but maybe a check is missing somewhere ?
Otherwise, we could consider `{| .. |}` strings to always be multiline ? That's how I consider them when I write code.

**Comment by gpetiot** [comment]
If we only consider the break before the closing parenthesis, it has indeed been fixed with #799, here is the new output:
```ocaml
let _ = f (fun () -> {|
Multi line
but short
|})

let _ =
  f (fun () ->
      {|
Multi line
with a lot of text
Foooooooooooooooooooooooooooooooooooooooooooooo
Bar
|})
```

**Comment by mbarbin** [comment]
Ah, sorry:  I was so focused on the paren change that I had originally missed that the formatting change shown by @Julow also contained a added break before the function f.  I was only referring to (and only thinking of) the paren in my statement about thinking it is related to #799 and feels like a bug. 

OK, let me try again now that I have seen there are 2 changes... :-)

Hmm.  I don't really understand how the size of the multi line affects the decision to break before the f, but the new output with the paren fix looks good to me.  My personal inclination is that I wouldn't consider it a priority and I'm basically happy to close that ticket if it's a tricky one.  Thanks!
